### DISCUSSION TITLE
ComboBox Popup Style

### DISCUSSION THREAD

**Discussion by somethingTacos** [AvaloniaUI/Avalonia#7571] [Q&A] [answered]
I can't seem to figure out how to change the popup style for the combobox to remove the white space at the top and bottom.

How would I go about changing the background of the combobox popup?
![image](https://user-images.githubusercontent.com/39627276/153245277-a2bb0f4e-957c-47da-86af-dcbf22307aaa.png)

```xaml
<Style Selector="ComboBox">
    <Setter Property="Foreground" Value="{StaticResource Foreground_Light}"/>
    <Setter Property="Background" Value="{StaticResource Background_Dark}"/>
  </Style>

  <Style Selector="ComboBox /template/ Path#DropDownGlyph">
    <Setter Property="Fill" Value="{StaticResource Foreground_Light}" />
  </Style>

  <Style Selector="ComboBox:pointerover /template/ Path#DropDownGlyph">
    <Setter Property="Fill" Value="{StaticResource Brush_Yellow}" />
  </Style>

  <Style Selector="ComboBox:pointerover">
    <Setter Property="Foreground" Value="{StaticResource Brush_Yellow}"/>
  </Style>

  <Style Selector="ComboBox:pointerover /template/ Border#Background">
    <Setter Property="Background" Value="Black"/>
    <Setter Property="BorderBrush" Value="Black" />
  </Style>

  <!-- I feel like isn't the correct element, but popup doesn't have a background property and popuproot does nothing -->
  <Style Selector="ComboBox /template/ Border#PopupBorder">
    <Setter Property="Background" Value="Black"/>
    <Setter Property="BorderBrush" Value="Black"/>
  </Style>

  <Style Selector="ComboBoxItem /template/ ContentPresenter">
    <Setter Property="Background" Value="Black"/>
  </Style>

  <Style Selector="ComboBoxItem">
    <Setter Property="Foreground" Value="{StaticResource Foreground_Light}"/>
    <Setter Property="Background" Value="{StaticResource Background_Dark}"/>
  </Style>
```
I've tried many combinations with other parts of the template, but I'm get the feeling I am doing something completely wrong.

I'm running `avalonia v0.10.12` incase it matters.

Any help is appreciated, as I'm basically at a lose on this one  :(

**Comment by amwx** [comment, marked as answer]
It seems the properties for the popup border are set directly on the Border with `{DynamicResource }`. There's a limitation with Avalonia's styling system right now that means these have style priority over everything else, so unfortunately you'll have to re-template the entire ComboBox to achieve what you want:

https://github.com/AvaloniaUI/Avalonia/blob/572e91b4a0f2fdcbe4d342bf90c2e07f3da0d2d9/src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml#L128-L132

**Reply by maxkatz6** [reply]
It's also possible to re-define these resource.

**Reply by somethingTacos** [reply]
Makes me sad I have to re-template the whole control for 1 background property, but that did work. Much appreciated :)

**Reply by timunie** [reply]
@somethingTacos see here: https://github.com/AvaloniaUI/Avalonia/discussions/7572#discussioncomment-2765796

**Comment by dubrovenskaya** [comment]
Now: ![2022-08-03 13_40_03-Window](https://user-images.githubusercontent.com/33326526/182599664-a191a4c4-1ee0-474f-9b00-b24c9296c070.png)
Wish: ![2022-08-03 13_37_44-Window](https://user-images.githubusercontent.com/33326526/182599685-63a8df5a-8f3a-4f48-8ade-5689340d39e2.png)
Hello together, may be someone can help me with the Idea how I can style the selected item and item bei hover? So, that it has special item background color and special item font weight. I started to learn Avalonia just vor 3 days... 

Best regards
Sascha

**Reply by timunie** [reply]
Hi Sascha, 

For Styles I always look up the original styles and copy the parts I need. In your case it can be found in here: https://github.com/AvaloniaUI/Avalonia/blob/4a6a3a141bfc94f34b7563e89eff463a36601cd2/src/Avalonia.Themes.Fluent/Controls/ComboBoxItem.xaml#L42-L47

Happy coding
Tim

**Comment by AnnyNeMy** [comment]
It  works for me: 
<Style Selector="ComboBox:dropdownopen /template/ Border#PopupBorder"><Setter Property="Background" Value="Red"></Setter><Setter Property="BorderBrush" Value="Red"></Setter> </Style>

**Reply by DirkDoes** [reply]
(a bit late to the converstation, however) This is the correct, and easiest way to do it.
The reason why it didnt work for you was because you didn't add `:dropdownopen`. Which is nessesairy for it to work
